libs/platform/ebpf_epoch.c

```diff
@@ -61,25 +68,28 @@ typedef struct _ebpf_epoch_state
     bool timer_disabled : 1; // Prevent re-arming the timer during shutdown.
 } ebpf_epoch_state_t;
 
+typedef struct _ebpf_epoch_thread_entry
+{
+    ebpf_epoch_state_t epoch_state;     // Per-thread epoch state.
+    uintptr_t old_thread_affinity_mask; // Thread affinity mask before entering an entry/exit block.
+    uint64_t last_used_time;            // Time when this entry was last used.
+} ebpf_epoch_thread_entry_t;
+
 // Table to track per CPU state.
 // This table must fit into a multiple of EBPF_CACHE_LINE_SIZE.
-typedef struct _ebpf_epoch_cpu_entry
+#pragma warning(disable : 4324) // Structure was padded due to alignment specifier.
+typedef __declspec(align(EBPF_CACHE_LINE_SIZE)) struct _ebpf_epoch_cpu_entry
 {
     ebpf_lock_t lock;
     _Guarded_by_(lock) ebpf_epoch_state_t epoch_state;                 // Per-CPU epoch state.
     _Guarded_by_(lock) ebpf_list_entry_t free_list;                    // Per-CPU free list.
     _Guarded_by_(lock) ebpf_hash_table_t* thread_table;                // Per-CPU thread table.
```

Comment:
Can the thread_table be replaced by a static array of 128 entries controlled by a counted semaphore?

Reply:
I have a prototype of the proposed changed. Going to mark this as draft, push an update and see if we hit any issues in CI/CD.